src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderSentence.kt

```diff
@@ -29,29 +37,187 @@ data class SentenceId(
 
 @Entity
 @Table(name = "OFFENDER_SENTENCES")
-// Warning: on mandatory fields mapped - just enough to create test entities
 data class OffenderSentence(
   @EmbeddedId
   val id: SentenceId,
   @OneToMany(mappedBy = "sentence", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
   val adjustments: MutableList<OffenderSentenceAdjustment> = mutableListOf(),
 
   @Column(name = "SENTENCE_STATUS")
-  val status: String,
+  val status: String, // 'I' or 'A'
+
   @ManyToOne(optional = false, fetch = LAZY)
   @JoinColumns(
     value = [
       JoinColumn(
         name = "SENTENCE_CALC_TYPE",
         referencedColumnName = "SENTENCE_CALC_TYPE",
-      ), JoinColumn(name = "SENTENCE_CATEGORY", referencedColumnName = "SENTENCE_CATEGORY"),
+      ),
+      JoinColumn(name = "SENTENCE_CATEGORY", referencedColumnName = "SENTENCE_CATEGORY"),
     ],
   )
   @BatchSize(size = 25)
   val calculationType: SentenceCalculationType,
+
   @Column(name = "START_DATE")
   val startDate: LocalDate,
+
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "ORDER_ID")
+  val courtOrder: CourtOrder? = null,
```

Comment:
interesting it is nullable - a sentence without a court order feels ILLEGAL

Reply:
yes - plenty of nulls. Will need to understand the meaning of no order Id, but for the mapping this is correct